### PR TITLE
Players now have names : >

### PR DESCRIPTION
**Pull request by marzsolt**

### CODE REVIEW

File: src/Client/Player/Player.py [hunk 1]
```diff
@@ -19,15 +19,26 @@ class PicFile(Enum):
...
 
-    def __init__(self, player_id, pic_file):
+    def __init__(self, player_id, name, pic_file):
         super(Player, self).__init__()
         self.logger = logging.getLogger('Domi.Player')
         try:
-            self.surf = pg.image.load(pic_file.value).convert()  # may have to change the path
+            img_surf = pg.image.load(pic_file.value).convert()  # may have to change the path
+            self.surf = pg.Surface((60, 60))
+            self.surf.fill((255, 253, 201))
```
Comment (DantElemer) [inline src/Client/Player/Player.py:28]: Khm... random numbers, which are coincidentally are the same as some lines below

File: src/Client/Player/Player.py [hunk 2]
```diff
@@ -19,15 +19,26 @@ class PicFile(Enum):
...
+            self.surf.blit(img_surf, (0, 10))
+            self.surf.set_colorkey((255, 253, 201),
+                pg.RLEACCEL)  # background color of the picture -> that color not shown
+
+            # showing player name
+            font = pg.font.SysFont('freesansbold', 17)
+            name_surf = font.render(name, False, WHITE)
+            name_surf_rect = name_surf.get_rect()
+            name_surf_rect.center = (self.surf.get_width() // 2, name_surf_rect.centery)
+            self.surf.blit(name_surf, name_surf_rect)
```
Comment (DantElemer) [inline src/Client/Player/Player.py:38]: Maybe add some comments?

File: src/Client/Player/PlayerManager.py [hunk 3]
```diff
@@ -21,16 +21,16 @@ def __init__(self):
...
+            self.players.append(ApplePlayer(player_id, names[player_id]))
         for player_id in orange_human_ids:
-            self.players.append(OrangePlayer(player_id))
+            self.players.append(OrangePlayer(player_id, names[player_id]))
         for player_id in apple_ai_ids:
-            self.players.append(ApplePlayer(player_id))
+            self.players.append(ApplePlayer(player_id, "A AI " + str(player_id)))
         for player_id in orange_ai_ids:
-            self.players.append(OrangePlayer(player_id))
+            self.players.append(OrangePlayer(player_id, "O AI" + str(player_id)))
```
Comment (DantElemer) [inline src/Client/Player/PlayerManager.py:33]: You could be more creative... :(

File: src/Server/Game/Game.py [hunk 4]
```diff
@@ -21,20 +21,22 @@ class Game:
...
+        self.__names = {}
         self.__player_logics = []
         self.__terrain = Terrain()
         self.running = True
 
     def update(self):
         self.__read_messages()
 
-        if not self.__game_started:
+        if not self.__game_started and not self.__game_start_signaled:
```
Comment (DantElemer) [inline src/Server/Game/Game.py:37]: Isn't it enough to check game signaled?

File: src/Server/Game/Game.py [hunk 5]
```diff
@@ -109,7 +111,8 @@ def __collect_players(self):
             mess = BaseMessage(sermess.MessageType.FIRST_PLAYER, sermess.Target.SCREEN)
             Server.get_instance().send_message(mess, self.__first_player_id)
         if len(connected_players) == self.__human_player_number:
-            self.__start_game()
+            Timer.sch_fun(15, self.__start_game, ())
+            self.__game_start_signaled = True
```
Comment (DantElemer) [inline src/Server/Game/Game.py:115]: What are you waiting for? Couldn't go on when you receive that? (And not wait a random time and hope it arrives during that.)

File: src/Client/Player/Player.py (hunk 1 above)
Comment (marzsolt) [reply src/Client/Player/Player.py:28]: Yep, but it just works. ;)

File: src/Client/Player/PlayerManager.py (hunk 3 above)
Comment (marzsolt) [reply src/Client/Player/PlayerManager.py:33]: :( I'm tired.

File: src/Client/Player/Player.py (hunk 1 above)
Comment (DantElemer) [reply src/Client/Player/Player.py:28]: Yes, until we change one of them...

**Pull request merged**